addons/website_sale/models/sale_order.py

```diff
@@ -442,7 +442,7 @@ def action_confirm(self):
         res = super().action_confirm()
         for order in self:
             if not order.transaction_ids and not order.amount_total and self._context.get('send_email'):
-                order._send_order_confirmation_mail()
+                order._send_payment_succeeded_for_order_mail()
```

Comment:
This is not related to a payment at all.
To investigate, but either this override can be dropped, or we have to keep the `_send_order_confirmation_email` logic.